### PR TITLE
Refactor calculations

### PR DESCRIPTION
**Pull request by tunamonster**
Refactored `calculations::extract_coordinates` and `calculations::coordinates_present?`

`coordinates_present?
`Checks whether an argument matches validating criteria (numeric, not NaN), rather than returning false when matching invalidating criteria.

Also replaced `.to_s == NaN` with` .to_f..nan? `

`extract_coordinates` 
Now calls `coordinates_present?` to validate the array, instead of comparing four booleans and initialising extra vars.

### CODE REVIEW

File: lib/geocoder/calculations.rb [hunk 1]
```diff
@@ -38,12 +38,7 @@ module Calculations
     # Returns true if all given arguments are valid latitude/longitude values.
     #
     def coordinates_present?(*args)
-      args.each do |a|
-        # note that Float::NAN != Float::NAN
-        # still, this could probably be improved:
-        return false if (!a.is_a?(Numeric) or a.to_s == "NaN")
-      end
-      true
+      args.map { |a| (a.is_a? Numeric and !a.to_f.nan?) ? true : false }.all?
```
Comment (lime) [inline lib/geocoder/calculations.rb]: Just a drive-by suggestion for simplifying this further:
  - The expression is already a boolean, the ternary `? true : false` is not needed
  - [`Enumerable#all?`](https://ruby-doc.org/core-2.3.1/Enumerable.html#method-i-all-3F) takes a block argument:

```ruby
args.all? { |a| a.is_a? Numeric and !a.to_f.nan? }
```

Comment (alexreisner) [inline lib/geocoder/calculations.rb]: @lime thanks for that, and I agree. @tunamonster will merge if you can make those two changes.

**Comment by alexreisner** [comment]
Just a heads-up that this caused some problems and had to be reverted. See #1130 for details.
